src/managers/quorum.py

```diff
@@ -30,6 +32,58 @@ class QuorumManager:
     def __init__(self, state: ClusterState):
         self.state = state
 
+    @cached_property
+    def client(self) -> ZooKeeperManager:
+        """Cached client manager application for performing ZK commands."""
+        admin_username = "super"
+        admin_password = self.state.cluster.internal_user_credentials.get(admin_username, "")
+        active_hosts = [server.host for server in self.state.started_servers]
+
+        return ZooKeeperManager(
+            hosts=active_hosts,
+            client_port=CLIENT_PORT,
+            username=admin_username,
+            password=admin_password,
+        )
+
+    @dataclass
+    class SyncStatus:
```

Comment:
**minor** I find it weird to have classes within classes. I know we can, but I have not found it very often. I would just move it outside of the class. 

It is not a strong point just a bit of preference. But I'm more curious to understand what are the use cases when you would do that and why?

Reply:
I agree, it usually seems a bit clunky to nest classes/functions.

My motivation here was to avoid endless tiny types, which makes mentally 'parsing' the code much more taxing. I felt that here, typing the Status to `SyncStatus` specifically, something that's referenced through `QuorumManager.SyncStatus` seemed logical to me.

I'd actually be fine moving it out if we need to, but that was the reasoning at least.